### PR TITLE
add pipeline file 

### PR DESCRIPTION
**Pull request by caitlinturnidge**
- Added a pipeline file that combines extract, load and rds_to_s3, this also contains a handler function for the Lambda.
- Moved the code in __name__ == "__main__" to a function above called extract(), so that I can move it to the pipeline file more easily.
- Added logging to the extract file and pipeline file. (In the extract I changed it to log when it got data for every 5 plants instead of every 1, we can change that if we want to, I just thought it was a lot of logs)
- Added a Dockerfile for when we want to make it into a lambda function.
- Moved test_extract.py into the pipeline folder.
- Created a new requirements.txt file with just the pipeline requirements, so that when we Dockerise it can be accessed.

### CODE REVIEW

File: pipeline/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM public.ecr.aws/lambda/python:latest
```
Comment (david-nepo) [inline pipeline/Dockerfile:1]: Dockerfile looks correct

File: pipeline/extract.py [hunk 2]
```diff
@@ -1,72 +1,66 @@
...
-
+import logging
 import requests
 import pandas as pd
 
 BASE_URL = 'https://data-eng-plants-api.herokuapp.com/plants/'
 NO_OF_PLANTS = 51
 
 
+def set_up_logger():
```
Comment (david-nepo) [inline pipeline/extract.py:10]: Logger looks correct, but add full stop to end of doc string for consistency.

File: pipeline/extract.py [hunk 3]
```diff
@@ -105,6 +96,9 @@ def get_origin_data(data: dict) -> dict:
             plant_data = get_plant_names(data)
             origin_data = get_origin_data(data)
 
+            if plant_id % 5 == 0 and plant_id > 0:
```
Comment (david-nepo) [inline pipeline/extract.py:99]: Nice oneee

File: pipeline/pipeline.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import logging
+from dotenv import load_dotenv
+
+from extract import extract
+from load import load
+from rds_to_s3 import update_rds_and_s3
+
+
+def set_up_logger():
+    """Set up a logger, to log pipeline progress to the console"""
```
Comment (david-nepo) [inline pipeline/pipeline.py]: Add full stop to doc string to keep consistent with the others.

File: pipeline/pipeline.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    logger.info("All plant data has been extracted and cleaned.")
+
+    load()
+    logger.info("Plant data has been loaded into the short term database.")
+
+    update_rds_and_s3()
+    logger.info("Old plant data has been moved to S3 storage.")
+
+
+def handler(event=None, context=None) -> None:
```
Comment (david-nepo) [inline pipeline/pipeline.py:32]: Handler syntax all looks correct, nice once

**Review by david-nepo — COMMENTED** [review]
All looks good to me - happy to approve

File: pipeline/pipeline.py (hunk 4 above)
Comment (caitlinturnidge) [reply pipeline/pipeline.py]: yep just changed them, thanks!

**Review by haydenjack — APPROVED** [review]
Looks good, happy to approve & merge

**Pull request merged**